pkg/injector/bootstrap.go

```diff
@@ -95,9 +104,125 @@ func (wh *mutatingWebhook) marshalAndSaveBootstrap(name, namespace string, confi
 			bootstrap.EnvoyXDSCACertFile:                  cert.GetTrustedCAs(),
 			bootstrap.EnvoyXDSCertFile:                    cert.GetCertificateChain(),
 			bootstrap.EnvoyXDSKeyFile:                     cert.GetPrivateKey(),
+			signingIssuerID:                               []byte(cert.GetSigningIssuerID()),
+			validatingIssuerID:                            []byte(cert.GetValidatingIssuerID()),
 		},
 	}
 
 	log.Debug().Msgf("Creating bootstrap config for Envoy: name=%s, namespace=%s", name, namespace)
 	return wh.kubeClient.CoreV1().Secrets(namespace).Create(context.Background(), secret, metav1.CreateOptions{})
 }
+
+// NewBootstrapSecretRotator returns a new bootstrap secret rotator.
+func NewBootstrapSecretRotator(kubeController k8s.Controller, certManager *certificate.Manager, checkInterval time.Duration) *BootstrapSecretRotator {
+	return &BootstrapSecretRotator{
+		kubeController: kubeController,
```

Comment:
We've recently put a lot of effort into hiding our k8s abstractions. Instead of giving this a kubeController, which propagates our leak of the k8s abstraction, can you give this a compute.Interface.

From there, let's add a new type 

```
type Secret struct {
 data map[string][]byte
}
```

Then we'll want to add methods to the compute.Interface (and kube.Client) to translate between the abstraction and the k8s client. So:

```
func (c *client) UpdateSecret(ctx context.Context, secret models.Secret) error {
   _, err := c.kubeController.UpdateSecret(ctx, secret)
  return err
}
```